shared/src/balancer_event_handler.rs

```diff
@@ -0,0 +1,154 @@
+use crate::{
+    current_block::BlockRetrieving,
+    event_handling::{BlockNumber, EventHandler, EventIndex, EventStoring},
+    impl_event_retrieving,
+    maintenance::Maintaining,
+};
+use anyhow::{anyhow, Context, Result};
+use contracts::{
+    balancer_v2_vault::{
+        self, event_data::PoolRegistered as ContractPoolRegistered, Event as ContractEvent,
+    },
+    BalancerV2Vault,
+};
+use ethcontract::common::DeploymentInformation;
+use ethcontract::{dyns::DynWeb3, Event as EthContractEvent, EventMetadata, H160};
+use std::ops::RangeInclusive;
+use tokio::sync::Mutex;
+
+#[derive(Debug)]
+pub enum BalancerEvent {
+    PoolRegistered(PoolRegistered),
+}
+
+#[derive(Debug, Default)]
+pub struct PoolRegistered {
+    pub pool_id: [u8; 32],
+    pub pool_address: H160,
+    pub specialization: u8,
+}
+
+#[derive(Default)]
+pub struct WeightedPool {
+    // pool_address: H160,
+// tokens: Vec<H160>,
+// pool_id:
+// TODO - other stuff
+}
+
+#[derive(Default)]
+pub struct BalancerPools {
+    // pools: HashMap<H160, HashSet<WeightedPool>>,
```

Comment:
It would be nice to get an idea what we will be storing here. and maybe populate that already. I assume the actual WeightedPools will be fetched lazily when needed and we will just store a mapping of token to pool addresses and maybe block number (for reorg handling)?

Reply:
Yes, this is more relevant to the next PR where we start populating this struct. I can shed a bit on light on it here though. 

I initially thoughts we would only need `HashMap<H160, HashSet<WeightedPool>>` (i.e. Tokens -> WeightedPools), but soon realized we also need to keep another `HashMap<PoolId, WeightedPool>` so we know if these pools have already been already been included. It may be possible that this is not needed, but I'd rather save this discussion for the following PR (https://github.com/gnosis/gp-v2-services/pull/642) still in Draft state.

Reply:
I'm also fine with leaving this empty for now and discussing the best approach in the following PR. As @fleupold suggested, we will likely need to keep some additional info in order to deal with re-orgs.